modules/ding_dashboard/ding_dashboard.module

```diff
@@ -0,0 +1,638 @@
+    'description' => t('Allow to push any editorial dashboard content.'),
+  );
+
+  $perm['approve own ed content'] = array(
+    'title' => t('Approve own editorial dashboard content.'),
+    'description' => t('Allow to push approve editorial dashboard content.'),
+  );
+
+  $perm['approve any ed content'] = array(
+    'title' => t('Approve any editorial dashboard content.'),
+    'description' => t('Allow to approve any editorial dashboard content.'),
+  );
+
+  $perm['reject own ed content'] = array(
+    'title' => t('Reject own editorial dashboard content.'),
+    'description' => t('Allow to reject own editorial dashboard content.'),
+  );
+
+  $perm['reject any ed content'] = array(
+    'title' => t('Reject own editorial dashboard content.'),
+    'description' => t('Allow to reject own editorial dashboard content.'),
+  );
+
+  $perm['delete own ed content'] = array(
+    'title' => t('Delete own editorial dashboard content.'),
+    'description' => t('Allow to delete own editorial dashboard content.'),
+  );
+
+  $perm['delete any ed content'] = array(
+    'title' => t('Delete any editorial dashboard content.'),
+    'description' => t('Allow to delete any editorial dashboard content.'),
+  );
+
+  return $perm;
+}
+
+/**
+ * Implements hook_secure_permissions().
+ */
+function ding_dashboard_secure_permissions($role) {
+  $permissions = array(
+    'guest blogger' => array(
+      'push own ed content',
+      'delete own ed content',
+    ),
+    'editor' => array(
+      'administer editorial dashboard',
+      'push own ed content',
+      'push any ed content',
+      'approve own ed content',
+      'approve any ed content',
+      'reject own ed content',
+      'reject any ed content',
+      'delete own ed content',
+      'delete any ed content',
+    ),
+    'local editor' => array(
+      'administer editorial dashboard',
+      'push own ed content',
+      'push any ed content',
+      'approve own ed content',
+      'approve any ed content',
+      'reject own ed content',
+      'reject any ed content',
+      'delete own ed content',
+      'delete any ed content',
+    ),
+    'local administrator' => array(
+      'administer editorial dashboard',
+      'push own ed content',
+      'push any ed content',
+      'approve own ed content',
+      'approve any ed content',
+      'reject own ed content',
+      'reject any ed content',
+      'delete own ed content',
+      'delete any ed content',
+    ),
+  );
+  if (isset($permissions[$role])) {
+    return $permissions[$role];
+  }
+}
+
+/**
+ * Access callback for editorial dashboard actions.
+ *
+ * @param object $node
+ *   Node object.
+ * @param string $action
+ *   Action name.
+ *
+ * @return array
+ *   Returns an array of permissions.
+ */
+function ding_dashboard_access_callback($node, $action) {
+  global $user;
+  $perms = &drupal_static(__FUNCTION__);
+  if (!isset($perms[$user->uid][$node->nid][$action])) {
+    switch ($action) {
```

Comment:
It seams that the check "$user->uid == $node->uid" could be move outside the switch as a fast check.